src/rp/index.ts

```diff
@@ -206,6 +219,51 @@ export class RelyingPartySolution {
     );
   }
 
+  /**
+   * Select a public key from those provided by the RP.
+   * Keys with algorithm "RSA-OAEP-256" or "RSA-OAEP" are expected, the firsts to be preferred.
+   *
+   * @param entity The RP entity configuration
+   * @returns A suitable public key with its compatible encryption algorithm
+   * @throws {NoSuitableKeysFoundInEntityConfiguration} If entity do not contain any public key suitable for encrypting
+   */
+  private choosePublicKeyToEncrypt(
+    entity: RpEntityConfiguration
+  ):
+    | { jwk: JWK & { alg: "RSA-OAEP-256" }; enc: "A256CBC-HS512" }
+    | { jwk: JWK & { alg: "RSA-OAEP" }; enc: "A128CBC-HS256" } {
+    const shuffle = () => (Math.random() > 0.5 ? 1 : -1);
+
+    // Look for keys using "RSA-OAEP-256", and pick a random one
+    const [usingRsa256] = entity.payload.jwks.keys
+      .filter(
+        <T>(k: T & { alg?: string }): k is T & { alg: "RSA-OAEP-256" } =>
+          typeof k.alg === "string" && k.alg === "RSA-OAEP-256"
+      )
+      .sort(shuffle);
```

Comment:
Why do you need it?
There is no need to pick a key randomly but by default just pick the first one that matches the requirement.

Reply:
also the way the sort function is written I'm taking either the first or the last one and never anything in between